sensor.py

```diff
@@ -0,0 +1,171 @@
+from homeassistant.const import EntityCategory, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.dt import utcnow
+
+from .const import DATA_GLINET, DOMAIN
+from .router import GLinetRouter
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class SystemStatusEntityDescription(SensorEntityDescription):
+    """Describes a GL-inet system status sensor entity."""
+
+    value_fn: Callable[[dict], int | float | None]
+
+
+SYSTEM_SENSORS: tuple[SystemStatusEntityDescription, ...] = (
+    SystemStatusEntityDescription(
+        key="cpu_temp",
+        name="cpu temperature",
+        has_entity_name=True,
+        icon="mdi:thermometer",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        value_fn=lambda system_status: system_status["cpu"]["temperature"],
+    ),
+    SystemStatusEntityDescription(
+        key="load_avg1",
+        name="Load avg (1m)",
+        has_entity_name=True,
+        icon="mdi:cpu-64-bit",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=2,
+        value_fn=lambda system_status: system_status["load_average"][0],
+    ),
+    SystemStatusEntityDescription(
+        key="load_avg5",
+        name="Load avg (5m)",
+        has_entity_name=True,
+        icon="mdi:cpu-64-bit",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=2,
+        value_fn=lambda system_status: system_status["load_average"][1],
+    ),
+    SystemStatusEntityDescription(
+        key="load_avg15",
+        name="Load avg (15m)",
+        has_entity_name=True,
+        icon="mdi:cpu-64-bit",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=2,
+        value_fn=lambda system_status: system_status["load_average"][2],
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up sensors."""
+    _LOGGER.debug("Setting up GL-inet Sensors")
+
+    router: GLinetRouter = hass.data[DOMAIN][entry.entry_id][DATA_GLINET]
+    sensors: list[SystemStatusSensor | SystemUptimeSensor] = [
+        SystemStatusSensor(router=router, entity_description=description)
+        for description in SYSTEM_SENSORS
+    ]
+    # Special case for uptime as it requires additional data processing
+    sensors.append(

```

Comment:
if `value_fn` evaluates to UNKNOWN from the get-go, we shouldn't add it as an entity.
Otherwise we generate clutter and issues:
![image](https://github.com/user-attachments/assets/567fa17a-8d08-47a5-baa8-da12d6df0e29)